tool/tctl/common/db_command_test.go

```diff
@@ -0,0 +1,189 @@
+	return slices.Clone(m.dbs), "", nil
+}
+
+func TestDBCommand_listDatabases(t *testing.T) {
+	db1 := mustCreateNewDatabase(t, "db1", "postgres", "localhost:5432", map[string]string{"env": "dev"})
+	db2 := mustCreateNewDatabase(t, "db2", "postgres", "localhost:5432", map[string]string{"env": "staging"})
+	db3 := mustCreateNewDatabase(t, "db3", "mysql", "localhost:5432", map[string]string{"env": "prod"})
+	db4 := mustCreateNewDatabase(t, "db4", "postgres", "localhost:5432", map[string]string{"env": "dev"})
+
+	db1Server1 := mustCreateDatabaseServerFromDB(t, db1, "server1")
+	db1Server2 := mustCreateDatabaseServerFromDB(t, db1, "server2")
+	db2Server1 := mustCreateDatabaseServerFromDB(t, db2, "server1")
+
+	// These don't exist in backend but used for validating output.
+	db3Server, err := toUnregisteredDBServer(db3)
+	require.NoError(t, err)
+	db4Server, err := toUnregisteredDBServer(db4)
+	require.NoError(t, err)
+
+	authClient := &mockAuthClientForDBCommand{
+		dbServers: []types.DatabaseServer{db1Server1, db1Server2, db2Server1},
+		dbs:       []types.Database{db1, db2, db3, db4},
+	}
+
+	tests := []struct {
+		name      string
+		dbCommand *DBCommand
+		wantDBs   []types.DatabaseServer
+		wantErr   bool
+	}{
+		{
+			name: "registered",
+			dbCommand: &DBCommand{
+				listRegistered: true,
+			},
+			wantDBs: []types.DatabaseServer{db1Server1, db1Server2, db2Server1},
+		},
+		{
+			name: "unregistered",
+			dbCommand: &DBCommand{
+				listUnregistered: true,
+			},
+			wantDBs: []types.DatabaseServer{db3Server, db4Server},
+		},
+		{
+			name: "registered and unregistered",
+			dbCommand: &DBCommand{
+				listRegistered:   true,
+				listUnregistered: true,
+			},
+			wantDBs: []types.DatabaseServer{db1Server1, db1Server2, db2Server1, db3Server, db4Server},
+		},
+		{
+			name: "unregistered with search keywords",
+			dbCommand: &DBCommand{
+				listUnregistered: true,
+				searchKeywords:   "mysql",
+			},
+			wantDBs: []types.DatabaseServer{db3Server},
+		},
+		{
+			name: "unregistered with predicate",
+			dbCommand: &DBCommand{
+				listUnregistered: true,
+				predicateExpr:    `name == "db4"`,
+			},
+			wantDBs: []types.DatabaseServer{db4Server},
+		},
+		{
+			name: "registered and unregistered with label",
+			dbCommand: &DBCommand{
+				listUnregistered: true,
+				listRegistered:   true,
+				labels:           "env=dev",
+			},
+			wantDBs: []types.DatabaseServer{db1Server1, db1Server2, db4Server},
+		},
+		{
+			name:      "no registered or unregistered",
+			dbCommand: &DBCommand{},
+			wantErr:   true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			actualDBs, err := tt.dbCommand.listDatabases(t.Context(), authClient)
+			if tt.wantErr {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+				require.ElementsMatch(t,
+					slices.Collect(types.ResourceNames(tt.wantDBs)),
+					slices.Collect(types.ResourceNames(actualDBs)),
+				)
+			}
```

Comment:
nit: to reduce indentation
```suggestion
			if tt.wantErr {
				require.Error(t, err)
				return
			}
			require.NoError(t, err)
			require.ElementsMatch(t,
				slices.Collect(types.ResourceNames(tt.wantDBs)),
				slices.Collect(types.ResourceNames(actualDBs)),
			)
```